seg_lapa/train.py

```diff
@@ -50,17 +57,37 @@ def test_step(self, batch, batch_idx):
         inputs, labels = batch
         outputs = self.model(inputs)
         loss = self.cross_entropy_loss(outputs, labels)
-
-        self.log("test_loss", loss, on_step=True, on_epoch=True, prog_bar=True, logger=True)
+        batch_loss = loss / len(batch[0])
+        wandb.log({"Test/Loss": batch_loss})
 
         return {
             "test_loss": loss,
         }
 
+    def training_epoch_end(self, outputs):
+        losses = torch.stack([x['loss'] for x in outputs])
+        gather_t = self.aggregate(losses)
+        gather_t = torch.flatten(torch.stack(gather_t))
+        loss = gather_t.mean()
```

Comment:
I believe we don't need to use dist. functions in <mode>_step_end methods. They should return a list of all the outputs from all the batches.
I'll verify this.

Reply:
if we have 2 GPUs, the function train_epoch_end will be invoked twice with the outputs argument of length batch_size

Reply:
Ah, OK.